tardis/utilities/asyncbulkcall.py

```diff
@@ -0,0 +1,173 @@
+from typing import TypeVar, Generic, Iterable, List, Tuple, Optional, Union
+from typing_extensions import Protocol
+import asyncio
+import time
+import sys
+
+from backports.cached_property import cached_property
+
+
+T = TypeVar("T")
+R = TypeVar("R")
+
+
+class BulkCommand(Protocol[T, R]):
+    """
+    Protocol of callables suitable for :py:class:`~.BulkExecution`
+
+    A bulk command must take an arbitrary number of tasks and is expected to provide
+    an iterable of one result per task. Alternatively, it may provide a single
+    :py:data:`None` value to indicate that there is no result. An unhandled
+    :py:class:`Exception` means that all tasks failed with that :py:class:`Exception`.
+    """
+
+    async def __call__(self, *__tasks: T) -> Optional[Iterable[R]]:
+        ...
+
+
+class AsyncBulkCall(Generic[T, R]):
+    """
+    Framework for queueing and executing several tasks via bulk commands
+
+    :param command: async callable that executes several tasks
+    :param size: maximum number of tasks to execute in one bulk
+    :param delay: maximum age in seconds of tasks before executing them
+    :param concurrent: how often the `command` may be executed at the same time
```

Comment:
This might be misleading as the concept of `bool` and `None` is not mentioned here. One must to read further explanations first to grasp the idea. But I currently have no better suggestion.

Reply:
I also do not get the point in having a `bool`and `None` here. Why is not sufficient to have just an `int`  and  a reasonable default?

Reply:
I've removed the ``bool`` cases. It now works the same as e.g. ``multiprocessing.Pool`` or ``concurrent.futures.ThreadPoolExecutor``: numbers are a fixed limit and ``None`` is "sensible default".

Reply:
Sounds good to me! Thanks!